### PR TITLE
clang-format 15.0.3

### PR DESCRIPTION
**Pull request by sleiner**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by H-G-Hristov** [comment]
Is this ready for review?

File: Formula/clang-format.rb [hunk 1]
```diff
@@ -83,6 +86,6 @@ def install
         shell_output("#{bin}/clang-format -style=Google test.c")
 
     ENV.prepend_path "PATH", bin
-    assert_match "test.c", shell_output("git clang-format")
+    assert_match "test.c", shell_output("git clang-format", 1)
```
Comment (SMillerDev) [inline Formula/clang-format.rb:89]: This doesn't seem correct. Why is it returning a non-zero code?

Comment (sleiner) [reply Formula/clang-format.rb:89]: Newer versions of `git-clang-format` do this to signal that some files were not formatted correctly (which is the case in this test). The corresponding [patch](https://github.com/llvm/llvm-project/commit/edbb8a843c130e60d71cb73e56a33d5ba2cc0ec9) landed in LLVM 14, but [due to a bug](https://github.com/llvm/llvm-project/issues/54758) which was only fixed in the 15.x release series, LLVM 15 is the first version to actually incorporate these return codes.

Comment (SMillerDev) [reply Formula/clang-format.rb:89]: Can we fix the file formatting instead?

Comment (sleiner) [reply Formula/clang-format.rb:89]: The file is formatted incorrectly on purpose (see also: line 79), as (as far as I can see) the point of the test is that (git-)clang-format recognizes and fixes the incorrect formatting.

Comment (SMillerDev) [reply Formula/clang-format.rb:89]: Ah right, I missed the `git` prefix. Looks good then

**Review by SMillerDev — APPROVED** [review]

**Comment by sleiner** [comment]
@SMillerDev How do we proceed? Is there anything I can/should do now? (Please pardon my ignorance; this is my first contribution to this project)

**Comment by SMillerDev** [comment]
Thanks @sleiner  ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Pull request closed without merging**